server/server/controllers/customer.js

```diff
@@ -30,4 +31,81 @@ async function getUsers(req, res) {
   res.json(users);
 }
 
-module.exports = { signup, getUsers };
+//get basket
+async function getBasket(req, res) {
+  const customerId = req.user.id; 
+  try {
+    // Retrieve the customer's basket
+    const basket = await Basket.findOne({ customerId }).populate("itemId.itemId");
+
+    if (!basket) {
+      return res.status(404).json({ message: "Basket not found" });
+    }
+
+    res.status(200).json({ message: "Basket retrieved successfully", basket });
+  } catch (error) {
+    console.error("Error retrieving basket:", error);
+    res.status(500).json({ message: "Internal server error" });
+  }
+}
+//add Item
+async function addItem(req, res) {
+  const { itemId, quantity } = req.body;
+  const customerId = req.user.id; 
+  try {
+    let basket = await Basket.findOne({ customerId });
+    if (!basket) {
+      basket = new Basket({ customerId });
+    }
+    basket.itemId.push({ itemId, quantity });
+    await basket.save();
+    
+    res.status(200).json({ message: "Item added to basket successfully", basket });
+  } catch (error) {
+    console.error("Error adding item to basket:", error);
+    res.status(500).json({ message: "Internal server error" });
+  }
+}
+//update item
+async function updateItem(req, res) {
+  const { itemId, quantity } = req.body;
+  const customerId = req.user.id; 
+  try {
+    let basket = await Basket.findOne({ customerId });
+
+    const itemIndex = basket.itemId.findIndex(id => id.toString() === itemId);
+
+    if (itemIndex === -1) {
+      return res.status(404).json({ message: "Item not found in basket" });
+    }
+    basket.itemId[itemIndex].quantity = quantity;
```

Comment:
the logic here is correct, but you should be using the correct schema and the correct fields names.